apollo-router/src/services/router/service.rs

```diff
@@ -812,12 +818,47 @@ impl RouterService {
         Ok(graphql_requests)
     }
 
-    fn count_errors(errors: &[graphql::Error]) {
+    fn count_errors(&self, errors: &[graphql::Error], context: &Context) {
+        let codes = errors
+            .iter()
+            .map(|e| e.extensions.get("code").and_then(|c| c.as_str()))
+            .collect();
+        self.count_error_codes(codes, context);
```

Comment:
Noting that this implementation will have to change if we add other dimensions here e.g. subgraph, coordinate, path - there could be some errors with the same codes from subgraph A vs. subgraph B, etc.